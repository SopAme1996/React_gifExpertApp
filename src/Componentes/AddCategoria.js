import React, { useState } from "react";
import PropTypes from "prop-types";

export const AddCategoria = ({ setData }) => {
  const [input, setInput] = useState("");

  const eventoTeclado = (e) => {
    setInput(e.target.value);
  };

  const eventoEnter = (e) => {
    e.preventDefault();

    if (input.trim().length > 2) {
      setData((data) => [input, ...data]);
      setInput("");
    }
  };

  return (
    <form onSubmit={eventoEnter}>
      <input
        type="text"
        value={input}
        onChange={eventoTeclado}
        placeholder="Ingrese su personaje favorito"
      />
    </form>
  );
};

AddCategoria.propTypes = {
  setData: PropTypes.func.isRequired,
};
