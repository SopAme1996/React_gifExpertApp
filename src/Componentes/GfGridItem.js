import React from "react";
import PropTypes from "prop-types";

export const GfGridItem = ({ title, url }) => {
  return (
    <div className="card">
      <img src={url} alt={title} />
      <p>{title}</p>
    </div>
  );
};

GfGridItem.propTypes = {
  title: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired,
};

export default GfGridItem;
