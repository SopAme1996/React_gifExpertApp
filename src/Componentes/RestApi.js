import React from 'react';
import { useFetchGifs } from '../Hooks/useFetchGifs';
import { GfGridItem } from './GfGridItem';
import PropTypes from 'prop-types'


export const RestApi = ({ categoria }) => {

    const {data:imagenes, loading}= useFetchGifs(categoria);

    return (
        <>
        <h3 className='animate__animated animate__fadeIn titulo'>{categoria}</h3>
        <hr className="line"/>
        <div className="animate__animated animate__backInLeft card-grid">

        {loading && <div className='containerLoading'><p>Loading....</p><div className='laoding'></div></div>}
        {
            imagenes.map(img => {
                return(
               <GfGridItem
                   key={img.id}
                   {...img}
              />
              )
            })
                }
        </div>
        </>
    )
}

RestApi.propTypes = {
categoria: PropTypes.string.isRequired
}