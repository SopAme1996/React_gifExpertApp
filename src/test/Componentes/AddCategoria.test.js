import  React from 'react';
import { AddCategoria } from '../../Componentes/AddCategoria';
import { shallow } from 'enzyme';

describe('Pruebas en el componente <AddCategoria/>', ()=>{
    
    const setData = jest.fn();
    let wrapper = shallow(<AddCategoria setData = {setData}/>);
    
    beforeEach(()=>{
        jest.clearAllMocks();
        wrapper = shallow(<AddCategoria setData = {setData}/>);
    });
    
    test('Debe mostrarse correctamente', () => {
        expect(wrapper).toMatchSnapshot();
    });

    test('debe de cambiar la caja de texto ', () => {
        const input = wrapper.find('input');
        const value = 'Hola Mundo';
        input.simulate('change', {target:{
            value: value
        }});
    });
    
    test('No debe de Â´postear la informacion onSubmit ', () => {
        wrapper.find('form').simulate('submit', {preventDefault(){}});

        expect(setData).not.toHaveBeenCalled();
    });

    test('Debe de llamar el setData y  limpiar la caja de texto ', () => {
        const input = wrapper.find('input');
        const data = 'Luis Enrique';

        input.simulate('change', {target:{
            value: data
        }})

        wrapper.find('form').simulate('submit', {preventDefault(){}});

        expect(setData).toHaveBeenCalled();
        expect(setData).toHaveBeenCalledWith( expect.any(Function) );
        expect(input.prop('value')).toBe('');
    })
    
})