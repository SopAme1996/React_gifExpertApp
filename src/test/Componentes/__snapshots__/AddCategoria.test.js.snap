// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente <AddCategoria/> Debe mostrarse correctamente 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    onChange={[Function]}
    placeholder="Ingrese su personaje favorito"
    type="text"
    value=""
  />
</form>
`;
